### ISSUE TITLE
Add H3 grid to "Create Grid" tool

### ISSUE THREAD

**Issue by tjukanovt** [qgis/QGIS#43531]
**Feature description.**
As the [H3 grid](https://github.com/uber/h3) is becoming more and more widely used, it would be nice to have support for that in QGIS.

**Additional context**
A logical place to have this would be IMHO in the Create Grid algorithm. It does already have support for hexagons, but those are not the "standard" H3 hexagons. Could be useful if the user could select the resolution and the extent.

We have initially discussed that @Joonalai could be interested in developing this feature. But all comments, questions and inputs are very much welcome.

Also this discussion could be helpful: https://gis.stackexchange.com/questions/310555/using-uber-h3-from-qgis and there is also a link to a PyQGIS implementation by @maphew --> https://github.com/maphew/code/blob/master/gis/qgis/h3-grid-from-layer.py

**Comment by Joonalai** [comment]
Indeed, as @tjukanovt  mentioned, I would be willing to join QGIS dev community and this would be a great feature to start the journey with!

**Comment by lbartoletti** [comment]
> Indeed, as @tjukanovt mentioned, I would be willing to join QGIS dev community and this would be a great feature to start the journey with!

Feel free to add the algorithm in `src/analysis/processing/qgsalgorithmgrid.cpp` with test file.

We can help you if you want.

And BTW, welcome :)

**Comment by Joonalai** [comment]
@lbartoletti the [PyQGIS script](https://github.com/maphew/code/blob/master/gis/qgis/h3-grid-from-layer.py) has an external dependency [h3-py](https://github.com/uber/h3-py), which is a binding of [H3 Core Library](https://github.com/uber/h3) written in C. What would be a best approach to apply an external H3 dependency?

**Comment by Joonalai** [comment]
@lbartoletti what do you think?

**Comment by arongergely** [comment]
Hi @Joonalai, I made an H3 plugin which adds grid creation  tools via an 'H3' processing provider. I just submitted it to the plugin repository today.

If you would like to take a look, here is the repo: 
https://github.com/arongergely/qgis-h3-toolkit-plugin

**Comment by hamiltoncj** [comment]
I just saw this. I have also implemented H3 in the Density Analysis plugin.
https://plugins.qgis.org/plugins/densityanalysis/

In this case I am using it to geohash points to H3 locations to do density analysis. Traditional methods would be to use the QGIS create grid (hexagon) and then do a count points in polygon, but with H3 this is orders of magnitude faster. 

I recommend H3 to be added to QGIS. If I were more into c++ and had a developers environment, I would implement some of the Density Analysis plugin code into core QGIS algorithms, but I don't have an environment set up to do this and my C++ is very rusty.

In the Density analysis plugin you can compare the difference in speed with the QGIS create grid method, vs geohash, vs H3 geohash.
